Assets/Scripts/SettingsInfo.cs

```diff
@@ -0,0 +1,253 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEditor.Search;
+using UnityEngine;
+using UnityEngine.Audio;
+using UnityEngine.Rendering;
+
+public struct Settings
+{
+    // Audio
+    public float masterVolume;
+    public float musicVolume;
+    public float sfxVolume;
+
+    // Graphics
+    public int qualityPresetIndex;
+    public int resolutionPresetIndex;
+    public int displayMode;
+
+    // Gameplay
+    public float sensScale;
+    public float playerFOV;
+    public float zoomFOV;
+}
+
+public class SettingsInfo : MonoBehaviour
+{
+    public static SettingsInfo Singleton { get; private set; }
+
+    private const string FileName = "/Settings.json";
+
+    private static string FilePath;
+    #region Graphics variables
+    public Resolution[] resolutions { get; private set; }
+
+    public string[] qualityNames { get; private set; }
+    #endregion
+    #region Audio variables
+    private const string audioGroupMaster = "masterVolume";
+    private const string audioGroupMusic = "musicVolume";
+    private const string audioGroupSFX = "sfxVolume";
+
+    private float maxVolumeMusic;
+    private float maxVolumeSFX;
+
+    [SerializeField]
+    private AudioMixer mainAudioMixer;
+    #endregion
+    #region Gameplay variables
+
+    [Header("Sensitivity Limits")]
+    public float lowerSensLimit = 0.1f;
+    public float upperSensLimit = 3f;
+
+    [Header("FOV Limits")]
+    public float lowerFOVLimit = 60f;
+    public float upperFOVLimit = 120f;
+
+    [Header("Zoom FOV Limits")]
+    public float lowerZoomFOVLimit = 20f;
+    public float upperZoomFOVLimit = 50f;
+
+    #endregion
+    public Settings settings = new()
+    {
+        sensScale = 1.0f,
+        masterVolume = 1.0f,
+        musicVolume = 1.0f,
+        sfxVolume = 1.0f,
+        qualityPresetIndex = 0,
+        resolutionPresetIndex = 0,
+        displayMode = 0,
+    };
+
+    private void Awake()
+    {
+        resolutions = Screen.resolutions.Reverse().ToArray();
+        qualityNames = QualitySettings.names;
+
+        mainAudioMixer.GetFloat(audioGroupMusic, out float musicVolume);
+        maxVolumeMusic = musicVolume;
+        mainAudioMixer.GetFloat(audioGroupSFX, out float sfxVolume);
+        maxVolumeSFX = sfxVolume;
```

Comment:
Hm, I wonder why the master volume wasn't treated the same way?

And for that matter, I'm not sure why this was necessary in the first place?

Reply:
FYI, it was necessary to prevent fading between music and sound to work, but it might be possible to just do this on master volume only